File: Codeforces Problems/Maximums.java
import java.util.*;

public class Maximums {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int i, j;
        long sum = 0, max = 0;

        long[] a = new long[n];

        for (i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }

        System.out.print(a[0] + " ");
        sum += a[0];

        for (i = 1; i < n - 1; i++) {
            sum += a[i];
            System.out.print(sum + " ");
            if (sum > max) {
                max = sum;
            }
        }

        if (sum == 0) {
            System.out.print(max + 1);
        } else {
            sum += a[i];
            System.out.print(sum);
        }

    }
}

File: Codeforces Problems/maximums.c
#include <stdio.h>
int main()
{
    long int n;
    scanf("%ld", &n);

    long long int a[n], i, j, sum = 0, max = 0;

    for (i = 0; i < n; i++)
    {
        scanf("%lld", &a[i]);
    }

    printf("%lld ", a[0]);
    sum = a[0];

    for (i = 1; i < n; i++)
    {
        if (a[i] < 0)
        {
            printf("%lld ", sum + a[i]);
        }
        else
        {
            sum += a[i];
            printf("%lld ", sum);
        }
    }

    return 0;
}